designs/default-zero-value.md

```diff
@@ -376,22 +357,96 @@ optional members that are not marked with `@default` or `@required` implicitly
 allow for null or omitted values.
 
 
-### Backward compatibility of the `@default` and `@required` trait
+## `@nullable` trait
+
+What is required today might not be required tomorrow. For cases when a service
+isn't sure if a member will be required forever, they can mark a `@required`
+member as `@nullable` to ensure that non-authoritative consumers of the model
+like clients treat the member as optional. The `@required` trait can be
+backward compatibly removed from a member marked as `@nullable` (and not
+replaced with the `@default` trait). This causes the `@required` trait to
+function as server-side validation rather than something that changes generated
+code.
+
+Structure members in Smithy are automatically considered nullable. For example,
+the following structure:
+
+```
+structure Foo {
+    baz: String
+}
+```
+
+Is equivalent to the following structure:
+
+```
+structure Foo {
+    @nullable
+    baz: String
+}
+```
+
+The primary use case of the `@nullable` trait is to indicate that while a
+member is _currently_ defined as `@required`, the service reserves the right to
+later remove the `@required` trait and make the member optional in the future.
+
+For example, `foo` in the following structure is considered a validation
+constraint rather than a type refinement trait:
 
-Backward compatibility rules of the `@default` and `@required` traits are as
-follows:
+```
+structure Foo {
+    @required
+    @nullable
+    foo: String
+}
+```
+
+The `@nullable` trait is defined in Smithy as:
+
+```
+@trait(
+    selector: "structure > member",
+    conflicts: [default]
+)
+structure default {}
+```
+
+The `@nullable` trait conflicts with the `@default` trait. The following
+model is invalid:
+
+```
+structure Message {
+    @default    // ERROR: this trait conflicts with @nullable.
+    @nullable
+    title: String
+}
+```
+
+
+## Backward compatibility rules
```

Comment:
These should be reflected in the evolving models guide.

Reply:
Done. I also rewrote the nullability specs to be, I hope, more clear and define backward compatibility in the spec instead of just in the non-normative guide.